pcsx2/DEV9/InternalServers/DNS_Logger.cpp

```diff
@@ -42,84 +42,87 @@ namespace InternalServers
 		LogPacket(&dns);
 	}
 
-	void DNS_Logger::LogPacket(DNS_Packet* dns)
+	std::string DNS_Logger::VectorToString(std::vector<u8> data)
 	{
-		Console.WriteLn("DEV9: DNS: ID %i", dns->id);
-		Console.WriteLn("DEV9: DNS: Is Response? %s", dns->GetQR() ? "True" : "False");
-		const char* opName;
-		switch ((DNS_OPCode)dns->GetOpCode())
+		if (data.size() != 0)
+		{
+			std::string str;
```

Comment:
You can utilize RVO / Copy elision ( https://en.cppreference.com/w/cpp/language/copy_elision ) if you declare the string outside the if-statement and always return it.
An empty string is equivalent to `""`